### ISSUE TITLE
[Bug]: popup - iframe stop working on LWC 3.8

### ISSUE THREAD

**Issue by josemvm** [3liz/lizmap-web-client#4863]
### What is the bug? (in English)

LWC 3.7.11
popup shows iframe in second table
![imagem](https://github.com/user-attachments/assets/e91e5528-cc3b-4f93-a4be-343fc97c9bec)

LWC 3.8.3-pre
popup doesn´t show iframe in second table
![imagem](https://github.com/user-attachments/assets/558c4278-85b4-42ce-a847-9a0f3abaeb55)

LWC 3.7.11 and LWV 3.8.3-pre
data - everything works fine
![imagem](https://github.com/user-attachments/assets/45326f3d-986f-498f-aa52-e5a00fb940c7)

### Steps to reproduce the issue

try beteween these versions

### Versions, safeguards, check summary etc

**Versions :**

* Lizmap Web Client : 3.8.3-pre.7994
* Lizmap plugin : 4.4.2
* QGIS Desktop : 3.34.11
* QGIS Server : 3.34.11
* Py-QGIS-Server : not used
* QGIS Server plugin atlasprint : 3.4.1
* QGIS Server plugin lizmap_server : 2.11.0
* QGIS Server plugin wfsOutputExtension : 1.8.2

<details>
<summary>List of Lizmap Web Client modules :</summary>
<br/>
* multiauth : 1.2.2
</details>

<details>
<summary>List of safeguards :</summary>
<br/>
* Mode : normal
* Allow parent folder : no
* Prevent other drive : no
* Prevent PG service : no
* Prevent PG Auth DB : no
* Force PG user&pass : no
* Prevent ECW : no
</details>

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### Operating system

Ubuntu 22.04

### Browsers

Firefox

### Browsers version

131.0.2

**Comment by josemvm** [comment]
this is the console output:

![imagem](https://github.com/user-attachments/assets/4fd0c792-a23d-4dca-84f7-1d2bbff6275d)

strange output... it's the same permissions to show the document in table (data tool) and as i said before on lwc 3.7 all works fine for popup tool and also for data tool

**Comment by josemvm** [comment]
https://github.com/3liz/lizmap-web-client/blob/fd79e9701f32ff63ba39390f65738b02af3628d1/lizmap/modules/view/controllers/media.classic.php#L106

```
    protected function error403($message)
    {
        /** @var jResponseJson $rep */
        $rep = $this->getResponse('json');
        $rep->data = array('error' => '403 forbidden (you\'re not allowed to access to this media)', 'message' => $message);
        $rep->setHttpStatus('403', 'Forbidden');

        return $rep;
    }

**Comment by Antoviscomi** [comment]
@josemvm possibly related to #4707 ?

**Comment by josemvm** [comment]
hi @Antoviscomi i'm talking about html, <iframe> tag

**Comment by Antoviscomi** [comment]
@josemvm that's now sanified as all dynamic container to avoid xss, so all the readdesing on dynamic contents shall be unavailable.

**Comment by josemvm** [comment]
>@josemvm that's now sanified as all html tags to dynamic container to avoid xss attacs, so all the readdresing on dynamic contents shall be unavailable.

@Antoviscomi yes i really understand the security issues but there should also be the possibility of creating exceptions for what is truly secure, i think

**Comment by Antoviscomi** [comment]
@josemvm I totally agree with you!

**Comment by josemvm** [comment]
![imagem](https://github.com/user-attachments/assets/480268b8-8056-461f-98bf-12ca949c9893)

**Comment by Antoviscomi** [comment]
@josemvm right but doesn't works without a parent layer, that is, if the layer that allows the iframe to be displayed does not have a parent or a relation setted I suppose.

**Comment by josemvm** [comment]
> @josemvm right but doesn't works without a parent layer

yes, but it's very strange...

**Comment by nboisteault** [comment]
Fixed by https://github.com/3liz/lizmap-web-client/pull/4953

**Comment by josemvm** [comment]
The transfer of “https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&project=operacoesU&path=media%2Fp_rusticos%2FR-ARL-C-2.pdf” was blocked because the iframe where it was triggered has the sandbox tag defined.

**Comment by josemvm** [comment]
the frame appears without content:

![Image](https://github.com/user-attachments/assets/57665cac-9a6c-45a8-9e38-26f1463cfd0b)

**Comment by Antoviscomi** [comment]
@josemvm not in mine as you can see below using 
```
<br>
		<p><iframe src=[%"n_link"%] width="540" height="700"></iframe></p>
</br>
```
and where `[%"n_link"%]` is the field containing the html file allable from my internal domain

https://github.com/user-attachments/assets/e5f066c1-f866-453b-a485-7a43513cba77

**Comment by josemvm** [comment]
@Antoviscomi what is your lizmap version?

**Comment by Gustry** [comment]
@josemvm You are just reopening this ticket by only copy/pasting your error, without context.

From your error :

> the iframe where it was triggered has the sandbox tag defined

Please search at least what `sandbox` means and share the code of the popup. Did you define `sandbox` in your popup ?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#sandbox

If sandbox is present, then it's more "secure" than "not present".

Like the iframe of @Antoviscomi which doesn't have the `sandbox` attribute.

**Comment by Antoviscomi** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) what is your lizmap version?

LWC 3.8.4, the same as your @josemvm

**Comment by Antoviscomi** [comment]
> [@josemvm](https://github.com/josemvm) You are just reopening this ticket by only copy/pasting your error, without context.
> 
> From your error :
> 
> > the iframe where it was triggered has the sandbox tag defined
> 
> Please search at least what `sandbox` means and share the code of the popup. Did you define `sandbox` in your popup ?
> 
> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#sandbox
> 
> If sandbox is present, then it's more "secure" than "not present".
> 
> Like the iframe of [@Antoviscomi](https://github.com/Antoviscomi) which doesn't have the `sandbox` attribute.

@Gustry @josemvm the behaviour it's correct  with `sandbox` attribute also, for me:
`<p><iframe src=[%"n_link"%] width="540" height="700" sandbox></iframe></p>`
in dev browser console I read:
`<iframe height="700" width="540" src="http://myhost/webgis/document/Norme_Attuative.html#AEA" sandbox="allow-scripts allow-forms"></iframe>` 
so I confirm there is not an issue in mine

**Comment by Gustry** [comment]
Yes, the `sandbox` atrribute can be added, but then it's up to the web server, (not related to Lizmap), to accept or not the request to serve the HTML content.

**Comment by Gustry** [comment]
Hum, not so sure, the `iframe` code can be analyzed from the webbrowser to see which attributes the sandbox can have after `DOMPurify`

https://github.com/3liz/lizmap-web-client/blob/7c5b679158c0101166ff020bb37bdc7f460ffb36/assets/src/modules/Utils.js#L214

**Comment by josemvm** [comment]
hi @Gustry and @Antoviscomi thanks for your help!

my error from firefox console:

```Download of “https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&project=operacoesU&path=media%2Fp_rusticos%2FR-GAFSAB-B-1.pdf” was blocked because the triggering iframe has the sandbox flag set.```

I don't have any `sandbox `attributes

i'm using `"popupSource": "lizmap"` like this:

```
<table>
<tr><td>Caderneta <i>Offline</i></td></tr>
<tr><td><iframe src="{$Caderneta Offline}" width="100%" height="500px"</iframe></td></tr>
</table>
```

on lwc 3.7.4 everything works fine!

**Comment by Antoviscomi** [comment]
> <tr><td>Caderneta <i>Offline</i></td></tr>
> <tr><td><iframe src="{$Caderneta Offline}"

@josemvm is Cardeneta Offline the file name? If so, can you try to rename it without spaces Cardeneta_Offline i.e.?

**Comment by josemvm** [comment]
> The lizmap popup for vector layer is deprecated, please use the button to transform to maptip.

if i transform to maptip the disappears from the map tree...

after conversion the popup, the html code will be in the .qgs file (?)
and into the cfg file it should be `"popupSource": "qgis"` and the template should be empty?

we have ore control over the template in the cfg file, like in tootip layers template

**Comment by Gustry** [comment]
> we have ore control over the template in the cfg file, like in tootip layers template

Sorry, I don't understand, what do you mean ? "more control" I guess, which control ?
QGIS Expressions are way more powerful.

**Comment by josemvm** [comment]
more control, i'm talking about editing html code

**Comment by Gustry** [comment]
Sorry @josemvm  I still don't understand, using a Lizmap popup for a vector layer is really discouraged : 

![Image](https://github.com/user-attachments/assets/de626421-3816-483a-a963-fe1499745435)

> more control, i'm talking about editing html code

All the HTML can be tweaked, either in QGIS HTML Maptip window.
When using a "QGIS HTML Maptip", have you tried this two buttons ?

![Image](https://github.com/user-attachments/assets/7ee4e244-48df-4fbc-a3f2-a867b45a5f17)

I'm curious by "more control about editing HTML code".

**Comment by Antoviscomi** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) the same result...
> 
> ```
> <br>
> 		<p><iframe src=[%"n_link"%] width="540" height="700"></iframe></p>
> </br>
> ```
> 
> your code comes from the .qgs file or .cfg file (popup template)?

My code came from QGIS html suggestion dialog, the is stored in .qgs

**Comment by josemvm** [comment]
@Gustry if the popup html code is stored in the cfg file, we can unescaoe the json code from template, edit the html code, escape to json code and paste again into the template - just this :-)

**Comment by josemvm** [comment]
@Antoviscomi what do you have in `"popupSource"` and in `"template"` in your cfg file template for your layer?

**Comment by Antoviscomi** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) what do you have in `"popupSource"` and in `"template"` in your cfg file for your layer?

`"popupSource": "auto",` and I don't find any (strictly) `"template"` term in my cfg

**Comment by Antoviscomi** [comment]
> hi [@Gustry](https://github.com/Gustry) and [@Antoviscomi](https://github.com/Antoviscomi) thanks for your help!
> 
> my error from firefox console:
> 
> `Download of “https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&project=operacoesU&path=media%2Fp_rusticos%2FR-GAFSAB-B-1.pdf” was blocked because the triggering iframe has the sandbox flag set.`
> 
> I don't have any `sandbox `attributes
> 
> i'm using `"popupSource": "lizmap"` like this template:
> 
> ```
> <table>
> <tr><td>Caderneta <i>Offline</i></td></tr>
> <tr><td><iframe src="{$Caderneta Offline}" width="100%" height="500px"</iframe></td></tr>
> </table>
> ```
> 
> on lwc 3.7.4 everything works fine!

@josemvm why you don't call the src as `src=Cardeneta Offline.pdf` ?  Is the file extension optional?

**Comment by josemvm** [comment]
@Gustry 
>The lizmap popup for vector layer is deprecated, please use the button to transform to maptip.
What is the result in the browser about the iframe code generated ?

a new project from scratch with a single layer and html map tip as:
`<iframe src="[%"caderneta_liz"%]" width="100%" height="500px"></iframe>`

the same result:
-  blank frame
-  error console `Download of “https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&project=popup&path=media%2Fp_rusticos%2FR-ARL-C-2.pdf” was blocked because the triggering iframe has the sandbox flag set.`
- html code `<iframe height="500px" width="100%" src="https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf" sandbox="allow-scripts allow-forms"></iframe>`

there is indeed a problem here compared to version 3.7.4.

@Gustry please try it yourself from scratch
thanks

**Comment by josemvm** [comment]
even adding `allow-same-origin` within webbrowser:

`<iframe height="500px" width="100%" src="https://gasur.cmarl.pt/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf" sandbox="allow-scripts allow-forms allow-same-origin"></iframe>`

**Comment by josemvm** [comment]
even adding allow-same-origin within my firefox webbrowser:

<iframe height="500px" width="100%" src="https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf" sandbox="allow-scripts allow-forms allow-same-origin"></iframe>

**Comment by josemvm** [comment]
the same for no `sandbox `attributes

`<iframe height="500px" width="100%" src="https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf"></iframe>`

**Comment by josemvm** [comment]
the same for just `allow-same-origin`  sandbox attribute

`<iframe height="500px" width="100%" src="https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf" sandbox="allow-same-origin"></iframe>`

thanks

**Comment by Antoviscomi** [comment]
> https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=popup&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf

@josemvm is this url reachable if you pasy it in browser?

**Comment by josemvm** [comment]
@Antoviscomi yes of course

**Comment by Gustry** [comment]
> even adding allow-same-origin within my firefox webbrowser:

Are you editing straight in the webbrowser ? I'm not sure that editing only, can trigger the webbrowser to understand the "new" sandbox attribute you are giving and try again to load the content.

I tried in #5256, no issue on my side as well :

```
<iframe src="[% "media" %]" width="100%" height="500px"></iframe>
```

The PDF is well displayed in FF 115.14.0esr. Using LWC master 3.10. The PR will be backported for the code refactoring done.

**Comment by josemvm** [comment]
>Are you editing straight in the webbrowser ?

@Gustry yes

**Comment by josemvm** [comment]
@Gustry what else can i do?... if when i switch to version 3.7.4 the `iframe `works as expected!

**Comment by Antoviscomi** [comment]
@Gustry @josemvm I tested iframe with .jpg and .html sources and it works as expected in 3.8.4, but the .pdf files are the problem for 3.8.4 only 

3.7.14 behaviour
https://github.com/user-attachments/assets/88baace4-45e5-41df-a02f-7f1f1588b0cb

3.8.4 behaviour

https://github.com/user-attachments/assets/0c5ed524-f771-4caa-9489-bc3df4b38328

same project same repository, opened with the two LWC different releases

**Comment by josemvm** [comment]
@Antoviscomi thanks
it's my case too!

**Comment by Antoviscomi** [comment]
@josemvm but seems a disallowing for pdfs, not a bug

**Comment by josemvm** [comment]
>but seems a disallowing for pdfs

but where!?...

**Comment by Antoviscomi** [comment]
> > but seems a disallowing for pdfs
> 
> but where!?...

As your can read in browser console message for 3.8.4 version, the javascript were locked by Chrome

**Comment by josemvm** [comment]
yes, it's true!

**Comment by josemvm** [comment]
is it blocked in one version (3.8.x) and not in another (3.7.x)!?...

**Comment by Antoviscomi** [comment]
> yes, it's true!

I think the only solution definitive is to implement a whithlist of IPs or Domains allowed to access the iframe, orherwise we'll forever affected from this behaviours

**Comment by josemvm** [comment]
my domain in not a public domain, it is within my local network

**Comment by Antoviscomi** [comment]
> my domain in not a public domain, it is within my local network

same for mine

**Comment by nboisteault** [comment]
I have not tested but if pdf files can be public you can put them in the root data folder above the repositories. You should be able to view them in iframes.

https://docs.lizmap.com/current/en/publish/configuration/media.html#use-a-single-media-folder-for-many-lizmap-folders

**Comment by Antoviscomi** [comment]
> I have not tested but if pdf files can be public you can put them in the root data folder above the repositories. You should be able to view them in iframes.
> 
> https://docs.lizmap.com/current/en/publish/configuration/media.html#use-a-single-media-folder-for-many-lizmap-folders

@nboisteault true for LWC 3.7.14 but non in 3.8.4 at the moment, i had put the pdf in a root (server) folder, i serve this root independently to LWC so this appear as cross-origin to the browser… that lock my domain only in LWC 3.8.4 as I tried to explain in the above comment

**Comment by Antoviscomi** [comment]
> [@nboisteault](https://github.com/nboisteault) true for LWC 3.7.14 but non in 3.8.4 at the moment, i had put the pdf in a root (server) folder, i serve this root independently to LWC so this appear as cross-origin to the browser… that lock my domain only in LWC 3.8.4 as I tried to explain in the above comment

@nboisteault and only for pdfs, I'm able to open .jpg, .html, .png and all other than pdfs

**Comment by Antoviscomi** [comment]
@Gustry @nboisteault @josemvm  is the `pdf-viewer` no more embedded  or `iframe` is no more able to call it in LWC 3.4.8? I ask you because of, as you can see in screencast below, 

https://github.com/user-attachments/assets/ee1756ae-b40b-43ec-b6f6-ea317ec81f4c

when I add the option `pdf-viewer-update-enabled` to the iframe def in browser console my pdf MAGICALLY works fine again

**Comment by Antoviscomi** [comment]
> https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/Structuring_content/General_embedding_technologies#the_embed_and_object_elements
> 
> > However, you are unlikely to use these elements very much. If you need to display PDFs, it's usually better to link to them, rather than embedding them in the page.

that's what I need to do, but in 3.7.14 this work normally despite of as It works (or better, doesn't works) in 3.8.4

**Comment by Antoviscomi** [comment]
I'm repeating myself but it only happens with pdfs and only in 3.8.4

**Comment by nboisteault** [comment]
We have made some changes in Lizmap 3.8.x to improve security against XSS. We'll accept a PR which improves PDF integration if it does not decrease security.

**Comment by Antoviscomi** [comment]
> We have made some changes in Lizmap 3.8.x to improve security against XSS. We'll accept a PR which improves PDF integration if it does not decrease security.

So it's insecure to call pdf-viewer on popup open or in dock open?

**Comment by josemvm** [comment]
@Antoviscomi can you send the `pdf-viewer-update-enabled` into the popup html code popup and make it works? i thing that you are using chrome, but im my case i can´t do it like you do...

@nboisteault any chance to recover this functionality?

**Comment by Antoviscomi** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) can you send the `pdf-viewer-update-enabled` into the popup html code popup and make it works? i thing that you are using chrome, but im my case i can´t do it like you do...

@josemvm both in Chrome as in FF if I try to pass pdf-viewer via Maptip it becomes sanitized/purified (I don't know why) I'm also trying to allow via Popup.js and Utils.js according to `DOMPurify.sanitize` in order to propose a PR but nothing! I also try to serve it by a node.js proxy server but nothing yet! The only way I found to temporarily view the pdf is via direct editing on html string in browser console! But my tests are in Progress

**Comment by josemvm** [comment]
@Antoviscomi thanks

**Comment by josemvm** [comment]
@Antoviscomi when i editing the html string in browser console this turns into:

`<iframe height="500px" width="100%" src="https://xxx.yyy.zzz/index.php/view/media/getMedia?repository=ma&amp;project=operacoesU&amp;path=media%2Fp_rusticos%2FR-ARL-C-2.pdf" sandbox="allow-scripts allow-forms" pdf-viewer-update-enabled=""></iframe>`

and doesn´t work...

**Comment by Gustry** [comment]
> I'm also trying to allow via Popup.js and Utils.js according to DOMPurify.sanitize in order to propose a PR but nothing

have you tried to build Javascript ? https://github.com/3liz/lizmap-web-client/blob/master/CONTRIBUTING.md#building-javascript-only
Or just build by using the Docker stack provided https://github.com/3liz/lizmap-web-client/blob/master/tests/README.md#with-linux

**Comment by Antoviscomi** [comment]
@Gustry yes of course but I dont understand why the pdf it seems yet not allowed

**Comment by nboisteault** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) can you send the `pdf-viewer-update-enabled` into the popup html code popup and make it works? i thing that you are using chrome, but im my case i can´t do it like you do...
> 
> [@nboisteault](https://github.com/nboisteault) any chance to recover this functionality?

Possibly but as usual it needs time, so funds or PRs are welcome.

**Comment by Antoviscomi** [comment]
> Possibly but as usual it needs time, so funds or PRs are welcome.

@nboisteault I'm trying to test an integration if it work I'll propose my PR

**Comment by Gustry** [comment]
> There is any step after js compilation I need to follow? I.e. php configurator.php?

This is for installation.

Have you tried the docker setup ? It's pretty straight forward.

**Comment by Antoviscomi** [comment]
> > There is any step after js compilation I need to follow? I.e. php configurator.php?
> 
> This is for installation.
> 
> Have you tried the docker setup ? It's pretty straight forward.

No, @Gustry I've not! 
but my tests seems to fail at the moment, there is something I don't yet understand on dhe beaviour.

**Comment by josemvm** [comment]
@Antoviscomi did you try something like this (certainly yes):

```
const cleanHTML = DOMPurify.sanitize('<iframe src="meuDoc.pdf" width="100%" height="500"></iframe>', {
  ALLOWED_TAGS: ['iframe'],
  ALLOWED_ATTR: ['src', 'width', 'height'],
  SANITIZE_DOM: true,
  AFTER_SANITIZATION: function(node) {
    // Check if the src is a valid PDF
    if (node.tagName === 'IFRAME' && node.src && node.src.endsWith('.pdf')) {
      // Allow PDF
      return true;
    }
    return false;
  }
});
```

**Comment by Antoviscomi** [comment]
@josemvm yes I tried i but the DOMPurify acts on every .pdf locking it when is passed by an url or a php.  I tried to create a new class pdf-frame passing it a server-side encoded pdf (on html page) but nothing! The CORS and the policy adopted to avoid XSS don't allow/permit any type of pdf or pdf/application to work in iframe or object. I'm very discouraged

**Comment by josemvm** [comment]
> So it's insecure to call pdf-viewer on popup open or in dock open?

i think so too

i think this is an overzealous approach to security within a popup or within a table

there are more pertinent security issues such as 2FA - it's just my opinion

we hope that 3liz team can fix this issue without compromising security

thanks to all

**Comment by Antoviscomi** [comment]
@Gustry @nboisteault @josemvm 
Finallyi think i found the only safe way to allow the display of pdfs in iframe. 
I try to explain how my idea should work so as to have your approval to globalize (at the moment works fine in my environment as in screencast below) it and propose it as PR.
After countless attempts i discovered that the only way we have to display the pdfs in the popup is to encode them in an html page through a php script so as to incorporate `<script src="https://cdnjs.cloudflare.com/ajax/libs/pdf.js/2.5.207/pdf.min.js"></script>`. 
In this way the popup allows the entry and display of the pdf without problems. my only doubt is: can i upload the additional php to integrate with the popup management? 
The test I did is based on a Postgis layer to allow the dynamic choice of encoded pdfs directly from QGIS html Maptip (necessary because the html pages also have the encoded name to avoid DOMPurify sanitizing them because of the .pdf in the extension). 
I have the idea of ​​making everything semi-automatic by leaving the admin the ability to choose the directory containing the pdfs and the postgis db within which the layer to link to the pdfs is located (important because the php scripts create a secure_id field within said file with the names of the new html outputs stored). 
So I ask you if this approach can be valid, otherwise I do not globalize it and avoid proposing it as PR.
Let me know.

https://github.com/user-attachments/assets/5c67b07a-361e-4e8e-ab2a-b16e31b777d0

**Comment by Antoviscomi** [comment]
@Gustry @nboisteault @rldhont I found another non invasive way to serve *.pdfs according with DOMPurify and LWC assets:
Integrate PDF viewer in Lizmap

- Added PDF viewer functionality in Utils.js
- Created pdfviewer.tpl to include PDF viewer container and navigation controls
- Added necessary dependencies (pdfjs-dist, dompurify) in package.json
- Defined and included whitelist in whitelist.js 

but this patch/integration it seems to introduce a  Client-Side URL Redirection's Vulnerability (in my Utils.js edition), Allowing unvalidated redirection based on user-specified URLs, so I ask you if is this vulnerability accettable, If so I propose this way as PR .

**Comment by Antoviscomi** [comment]
Fix popup - iframe stop working on LWC 3.8 #4863 by Integrate PDF viewer in Lizmap #5408

**Comment by Antoviscomi** [comment]
> [@Antoviscomi](https://github.com/Antoviscomi) did this fix solve your issue?

I don't know @josemvm my PR is still under review

**Comment by josemvm** [comment]
@Antoviscomi yes i'm talking about with this fix https://github.com/3liz/lizmap-web-client/pull/5414
the issue has been closed by @nboisteault but the problem remains

**Comment by Antoviscomi** [comment]
@josemvm because of in this issue they are two different bugs, the iframes are reallowed the the main issue is solved, but without pdfs

**Comment by Gustry** [comment]
As I said in #5411 

With latest 3.8, PDF works for me :
Layer `iframe_pdf`, the single red point on the map.

https://sandbox.lizmap.com/lizmap_3_8/index.php/view/map?repository=tests&project=xss

I tried on both Chromium and Firefox.

Is-it still not working for you ?

**Comment by Antoviscomi** [comment]
> As I said in [#5411](https://github.com/3liz/lizmap-web-client/pull/5411)
> 
> With latest 3.8, PDF works for me : Layer `iframe_pdf`, the single red point on the map.
> 
> https://sandbox.lizmap.com/lizmap_3_8/index.php/view/map?repository=tests&project=xss
> 
> I tried on both Chromium and Firefox.
> 
> Is-it still not working for you ?

Thanks @Gustry @nboisteault yes with the 3.8.6-pre it works with sandbox attribute also!

**Comment by Antoviscomi** [comment]
@josemvm I think yoou can close this issue now

**Comment by josemvm** [comment]
thank you very very much to all 👍
